presto-main/src/test/java/io/prestosql/type/TestUUIDType.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.type;
+
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.block.BlockBuilder;
+import org.testng.annotations.Test;
+
+import static io.prestosql.type.UuidType.UUID;
+import static org.testng.Assert.assertEquals;
+
+public class TestUUIDType
+        extends AbstractTestType
+{
+    public TestUUIDType()
+    {
+        super(UUID, String.class, createTestBlock());
+    }
+
+    public static Block createTestBlock()
+    {
+        BlockBuilder blockBuilder = UUID.createBlockBuilder(null, 1);
+        UUID.writeSlice(blockBuilder, getSliceForUUID("6b5f5b65-67e4-43b0-8ee3-586cd49f58a0"));
```

Comment:
This could be simplified a bit:
```java
for (int i = 0; i < 10; i++) {
    UUID.writeSlice(Slices.wrappedLongArray(0x6B5F5B6567E443B0, 0x8EE3586CD49F58A0 + i);
}
```
This also makes it clear that the value is changing (you need to look hard for the last character)

Or if you think it's better with text:
```java
for (int i = 0; i < 10; i++) {
    String uuid = "6b5f5b65-67e4-43b0-8ee3-586cd49f58a" + i
    UUID.writeSlice(castFromVarcharToUuid(utf8slice(uuid)));
}
```